js-core/homeworks/homework-17/index.js

```diff
@@ -0,0 +1,117 @@
+// TASK 0
+const solution = arr => {
+	let newArr = [];
```

Comment:
For at all, we could look at what we need from our array? We have to reduce its size but leave the same arguments.
So the first step will be to use .filter method.


A solution can look that way
1) Take the element
2) Take the element-index
3) Take leaved part of an array to element-index
4) Check if **some** elemnt in array is bigger than current element ( please note, there the same JavaScript method called .some
5) if no one from that elements is bigger than current, leave it in array otherwise drop it

So I guess it could be solved with

.filter + .some from arrays method